### ISSUE TITLE
Support additional string formatting methods

### ISSUE THREAD

**Issue by brocksam** [brocksam/pyproprop#17]
Support formatting of strings to other cases such as camel and snake as well as upper, lower and title. This will mainly involve handling replacement of spaces and dealing with invalid characters. Upper cases used in abbreviations/acronyms will also need to be handled intelligently.

Examples:
```python
some_string = "Some String with ABRV."
format_snake(some_string)
> "some_string_with_abrv"
format_camel(some_string)
> "SomeStringWithABRV"
```

**Comment by brocksam** [comment]
Added in 212bc909631519149a7e732fa75f4d34071191ee, d5a6f295bd0ab387d0fa536eeb361257546dda25, 6f7edb1485dd434e103653bba39b7455bac94965 and 93f67e898e693c4db9372d18d88b67ba7af61d7a.

**Comment by brocksam** [comment]
Resulting bugs fixed in 6bc0dcf0ede4d3a2e5cbcf30c7c7af62a587199e, fe90699ec047e84902005863751b98c748c9af59, 160f36d54e43f1b438c1658447e40676157c27a9, a053dff5cfaf282effd014adb6f25e7a82040d38, cfdbdadc70aba37d15e81104d98e8a9154acea67, 5599cde2c7bf415b33ae893d2d9a6c1a5e34801a and 2b62bd55bb3503e7aa9e640d58ba139638c35ca4.
